homeassistant/components/lupusec/__init__.py

```diff
@@ -39,44 +41,77 @@
     extra=vol.ALLOW_EXTRA,
 )
 
-LUPUSEC_PLATFORMS = [
+LUPUSEC_PLATFORMS: list[Platform] = [
     Platform.ALARM_CONTROL_PANEL,
     Platform.BINARY_SENSOR,
     Platform.SWITCH,
 ]
 
 
-def setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up Lupusec component."""
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the lupusec integration."""
+
+    if DOMAIN not in config:
+        return True
+
     conf = config[DOMAIN]
-    username = conf[CONF_USERNAME]
-    password = conf[CONF_PASSWORD]
-    ip_address = conf[CONF_IP_ADDRESS]
+    username = conf.get(CONF_USERNAME)
+    password = conf.get(CONF_PASSWORD)
+    host = conf.get(CONF_IP_ADDRESS)
     name = conf.get(CONF_NAME)
 
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_USERNAME: username,
+                CONF_PASSWORD: password,
+                CONF_HOST: host,
+                CONF_FRIENDLY_NAME: name,
+            },
+        )
+    )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up this integration using UI."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    host = entry.data.get(CONF_HOST)
+    username = entry.data.get(CONF_USERNAME)
+    password = entry.data.get(CONF_PASSWORD)
+    name = entry.data.get(CONF_NAME)
+
     try:
-        hass.data[DOMAIN] = LupusecSystem(username, password, ip_address, name)
-    except LupusecException as ex:
-        _LOGGER.error(ex)
-
-        persistent_notification.create(
-            hass,
-            f"Error: {ex}<br />You will need to restart hass after fixing.",
-            title=NOTIFICATION_TITLE,
-            notification_id=NOTIFICATION_ID,
+        lupusec_system = await hass.async_add_executor_job(
+            LupusecSystem, username, password, host, name
         )
+    except LupusecException:
+        _LOGGER.error("Failed to connect to Lupusec device at %s", host)
         return False
+    except Exception as ex:  # pylint: disable=broad-except
+        _LOGGER.error(
+            "Unknown error while trying to connect to Lupusec device at %s: %s",
+            host,
+            ex,
+        )
+        return False
+
+    hass.data[DOMAIN] = lupusec_system
```

Comment:
```suggestion
    hass.data.setdefault(DOMAIN, {}) = lupusec_system
```

Also, talking about the single integration allowed, this is where you can fix this.

```suggestion
    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = lupusec_system
```

This would allow the user to set more integrations